### ISSUE TITLE
Using ignore_empty_think breaks reasoning process during fine-tuning

### ISSUE THREAD

**Issue by JhaceLam** [modelscope/ms-swift#5927]
I tried to finetune Qwen3-4B-Thinking with the following command:

```bash
megatron sft \
    --load Qwen3-4B-Thinking-2507-mcore \
    --dataset data/sft_data_empty_think.jsonl \
    --loss_scale ignore_empty_think \
    --finetune true \
    --cross_entropy_loss_fusion true \
    --lr 1e-5 \
    --lr_warmup_fraction 0.05 \
    --min_lr 1e-6 \
    --max_epochs 1 \
    --save megatron_output/Qwen3-4B-Thinking-2507 \
```

The dataset format (sft_data_empty_think.jsonl) looks like this:
```json
{"messages": [
    {"role": "system", "content": "<system-prompt>"},
    {"role": "user", "content": "<query>"},
    {"role": "assistant", "content": "<think>\n\n</think>\n\nResults are xxxxxxx"}
]}
```

However, after fine-tuning, when I run inference on a test set in the same format, the model outputs responses like:
```json
{"response": "<think>\nResults are xxxxxxx"}
```
The reasoning process structure is broken—the <think> tag is not closed properly, and the answer is merged into the reasoning section.

Could you please give me some adivce?

**Comment by Jintao-Huang** [comment]
I recommend mixing in some "thinking" data; you can use a distilled dataset.
